### PR TITLE
changes for dplyr 1.0.0

### LINKED ISSUE #117: dplyr 1.0.0 rev dep checks
We see this when testing `tidygraph`: 

```
[master] 74.5 MiB ❯ revdepcheck::revdep_details(revdep = "tidygraph")
══ Reverse dependency check ═══════════════════════════════════════════════ tidygraph 1.1.2 ══

Status: BROKEN

── Newly failing

x checking tests ...
x checking S3 generic/method consistency ... WARNING
x checking Rd cross-references ... WARNING

── Before ────────────────────────────────────────────────────────────────────────────────────
0 errors ✓ | 0 warnings ✓ | 0 notes ✓

── After ─────────────────────────────────────────────────────────────────────────────────────
> checking tests ...
  See below...

> checking S3 generic/method consistency ... WARNING
  pull:
    function(.data, var, name)
  pull.morphed_tbl_graph:
    function(.data, var)
  
  pull:
    function(.data, var, name)
  pull.tbl_graph:
    function(.data, var)
  
  See section ‘Generic functions and methods’ in the ‘Writing R
  Extensions’ manual.

> checking Rd cross-references ... WARNING
  Missing link or links in documentation object 'graph_join.Rd':
    ‘join.tbl_df’
  
  See section 'Cross-references' in the 'Writing R Extensions' manual.

── Test failures ─────────────────────────────────────────────────────────────── testthat ────

> library(testthat)
> library(tidygraph)

Attaching package: 'tidygraph'

The following object is masked from 'package:testthat':

    matches

The following object is masked from 'package:stats':

    filter

> 
> test_check("tidygraph")
── 1. Failure: bind_nodes works (@test-bind.R#19)  ─────────────────────────────
as_tibble(gr1) not equal to `tbl`.
Attributes: < Component "class": Lengths (3, 1) differ (string compare on first 1) >
Attributes: < Component "class": 1 string mismatch >
Attributes: < Component "row.names": Modes: numeric, character >
Attributes: < Component "row.names": target is numeric, current is character >

══ testthat results  ═══════════════════════════════════════════════════════════
[ OK: 274 | SKIPPED: 0 | WARNINGS: 3 | FAILED: 1 ]
1. Failure: bind_nodes works (@test-bind.R#19) 

Error: testthat unit tests failed
Execution halted

1 error x | 2 warnings x | 0 notes ✓
```

### PR DESCRIPTION
**Pull request by romainfrancois**
closes #117

### CODE REVIEW

File: R/pull.R [hunk 1]
```diff
@@ -1,15 +1,17 @@
 #' @export
 #' @importFrom dplyr pull
-pull.tbl_graph <- function(.data, var = -1) {
+pull.tbl_graph <- function(.data, var = -1, name = NULL) {
   d_tmp <- as_tibble(.data)
   var <- enquo(var)
-  pull(d_tmp, !! var)
+  name <- enquo(name)
+  pull(d_tmp, !! var, !! name)
```
Comment (romainfrancois) [inline R/pull.R:7]: might be replaced by `pull(d_tmp, {{var}}, {{name}})`

File: R/pull.R [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 }
 #' @export
 #' @importFrom dplyr pull
-pull.morphed_tbl_graph <- function(.data, var) {
+pull.morphed_tbl_graph <- function(.data, var, name = NULL) {
   var <- enquo(var)
-  lapply(.data, pull, !! var)
+  name <- enquo(var)
+  lapply(.data, pull, !! var, !! name)
```
Comment (romainfrancois) [inline R/pull.R:14]: Does this work, to use `!!` in `lapply()` ?

File: tests/testthat/test-bind.R [hunk 3]
```diff
@@ -16,7 +16,7 @@ test_that('bind_nodes works', {
   gr1 <- tbl_graph(head(mtcars))
   gr1 <- bind_nodes(gr1, tail(mtcars))
   tbl <- dplyr::bind_rows(head(mtcars), tail(mtcars))
-  expect_equal(as_tibble(gr1), tbl)
+  expect_equivalent(as_tibble(gr1), tbl)
```
Comment (romainfrancois) [inline tests/testthat/test-bind.R:19]: or `as_tibble(tbl)`, the only problem is that `tbl` is a data frame

File: R/pull.R (hunk 2 above)
Comment (thomasp85) [reply R/pull.R:14]: yeah... the expressions are lazily passed on to `fun`, so yeah

File: R/pull.R (hunk 1 above)
Comment (thomasp85) [reply R/pull.R:7]: yeah... this is fine. I haven't double-braced the tidy eval code in tidygraph so this fits the rest of the code

File: R/pull.R (hunk 2 above)
Comment (romainfrancois) [reply R/pull.R:14]: so perhaps also `{{var}}` and `{{name}}` i.e. without doing the `enquo()` first

Comment (thomasp85) [reply R/pull.R:14]: maybe, but as said above your current fix is in line with the rest of the codebase

**Comment by thomasp85** [comment]
Thanks

**Pull request merged**